### PR TITLE
Event and Context updates

### PR DESCRIPTION
**Pull request by cwallsfdc**
Updates per programming model changes.  See https://github.com/forcedotcom/sf-fx-sdk-nodejs/pull/44.

Should we bump to version 1.0.0?
`"version": "0.0.1",`

### CODE REVIEW

File: middleware/index.ts [hunk 1]
```diff
@@ -1,69 +1,185 @@
...
+    if (!userContext) {
+        const message = `UserContext not provided: ${JSON.stringify(reqContext)}`;
+        throw new Error(message);
+    }
+
+    const apiVersion = reqContext.apiVersion || process.env.FX_API_VERSION || Constants.CURRENT_API_VERSION;
+    const user = createUser(userContext);
+
+    // If accessToken was provided, setup APIs.
+    let dataApi: DataApi;
```
Comment (dhagberg-sf) [inline middleware/index.ts]: I think these need to be of type `DataApi | undefined` and `UnitOfWork | undefined` to be able to be compiled strict.

File: middleware/package.json [hunk 2]
```diff
@@ -25,7 +26,7 @@
   },
   "author": "TODO",
   "dependencies": {
-    "@salesforce/salesforce-sdk": "0.0.1-develop.12",
+    "@salesforce/salesforce-sdk": "1.0.0",
```
Comment (dhagberg-sf) [inline middleware/package.json]: yay!

File: .vscode/launch.json [hunk 3]
```diff
@@ -4,6 +4,18 @@
     // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
     "version": "0.2.0",
     "configurations": [
+        {
```
[the remark this answers is not in the document]
Comment (cwallsfdc) [reply .vscode/launch.json]: It's convenient to share run config, ie mocha test all.

File: buildpack.toml [hunk 4]
```diff
@@ -3,7 +3,7 @@ api = "0.2"
 [buildpack]
 id = "salesforce/nodejs-fn"
 name = "Salesforce NodeJS Function Structured Middleware Buildpack"
-version = "0.0.12"
+version = "1.0.0"
```
Comment (dhagberg-sf) [inline buildpack.toml]: 🚢

**Pull request merged**